plugins/modules/pacemaker_resource.py

```diff
@@ -0,0 +1,231 @@
+    default: present
+    type: str
+  name:
+    description:
+      - Specify the resource name to create.
+    required: true
+    type: str
+  resource_type:
+    description:
+      - Resource type to create.
+    type: dict
+    suboptions:
+      resource_name:
+        description:
+          - Specify the resource type name.
+        type: str
+      resource_standard:
+        description:
+          - Specify the resource type standard.
+        type: str
+      resource_provider:
+        description:
+          - Specify the resource type providers.
+        type: str
+  resource_option:
+    description:
+      - Specify the resource option to create.
+    type: list
+    elements: str
+    default: []
+  resource_operation:
+    description:
+      - List of operations to associate with resource.
+    type: list
+    elements: dict
+    default: []
+    suboptions:
+      operation_action:
+        description:
+          - Operation action to associate with resource.
+        type: str
+      operation_option:
+        description:
+          - Operation option to associate with action.
+        type: list
+        elements: str
+  resource_meta:
+    description:
+      - List of meta to associate with resource.
+    type: list
+    elements: str
+  resource_argument:
+    description:
+      - Action to associate with resource.
+    type: dict
+    suboptions:
+      argument_action:
+        description:
+          - Action to apply to resource.
+        type: str
+        choices: [ clone, master, group, promotable ]
+      argument_option:
+        description:
+          - Options to associate with resource action.
+        type: list
+        elements: str
+  wait:
+    description:
+      - Timeout period for polling the resource creation.
+    type: int
+    default: 300
+'''
+
+EXAMPLES = '''
+---
+- name: Create pacemaker resource
+  hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: Create virtual-ip resource
+    community.general.pacemaker_resource:
+      state: present
+      name: virtual-ip
+      resource_type:
+        resource_name: IPaddr2
+      resource_option:
+        - "ip=[192.168.2.1]"
+      resource_argument:
+        argument_action: group
+        argument_option:
+          - master
+      resource_operation:
+        - operation_action: monitor
+          operation_option:
+            - interval=20
+'''
+
+RETURN = '''
+cluster_resources:
+    description: The cluster resource output message.
+    type: str
+    sample: "Assumed agent name ocf:heartbeat:IPaddr2 (deduced from IPaddr2)"
+    returned: always
+'''
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.pacemaker import pacemaker_runner
+
+
+class PacemakerResource(StateModuleHelper):
+    change_params = ('value', )
+    diff_params = ('value', )
+    output_params = ('status', 'value')
```

Comment:
Errr, I see how I can have created some confusion, and I will make sure to update the ModuleHelper guide to cover that point: these variables should only be set for _module parameters_, which neither `status` nor `value` are. I reckon in this module we can get rid of the 3 variables entirely.